src/main/scala/io/scalac/wtf/domain/User.scala

```diff
@@ -24,45 +25,40 @@ case class NewUser(email: String, password: String)
 object User {
 
   sealed trait ValidationError
-  final case object WrongEmailPattern extends ValidationError
-  final case object PasswordTooShort  extends ValidationError
-  final case object UserAlreadyExists extends ValidationError
+  case object WrongEmailPattern extends ValidationError
+  case object PasswordTooShort  extends ValidationError
+  case object UserAlreadyExists extends ValidationError
 
   private val emailPattern = "^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}$".r
 
   def validateUser(email: String, password: String)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], User]] = {
-    val validatedUser = for {
-      emailV <- validateEmail(email).toValidated
-      emailNel = emailV.toValidatedNel
-      passwordNel = validatePassword(password).toValidatedNel
-    } yield (emailNel |@| passwordNel) map { 
-      case (email, password) => User(email = email, password = password)
+
+    val validation = validateEmail(email).toValidated map { emailV =>
+      (emailV.toValidatedNel |@| validatePassword(password).toValidatedNel) map (User.apply(None, _, _))
```

Comment:
I can see your point, but I think these two versions are a matter of personal preference.
My version has more lines but the lines are more straightforward. And I do like for-comprehensions :)

You version is shorter no doubt about it.

I wonder what @pjazdzewski1990 thinks about it.